app/invites/server/functions/validateInviteToken.js

```diff
@@ -0,0 +1,30 @@
+import { Meteor } from 'meteor/meteor';
+
+import { Invites, Rooms } from '../../../models';
+
+export const validateInviteToken = (userId, token) => {
+	if (!userId) {
+		throw new Meteor.Error('error-invalid-user', 'The user is invalid', { method: 'validateInviteToken', field: 'userId' });
+	}
+	if (!token) {
+		throw new Meteor.Error('error-invalid-token', 'The invite token is invalid.', { method: 'validateInviteToken', field: 'token' });
+	}
+
+	const inviteData = Invites.findOneByHash(token);
+	if (!inviteData) {
+		return {
+			valid: false,
+		};
+	}
+
+	const room = Rooms.findOneById(inviteData.rid);
+	if (!room) {
+		return {
+			valid: false,
+		};
+	}
+
+	return {
+		valid: true,
+	};
```

Comment:
Could return the room and the inviteData here and reuse it at `app/invites/server/functions/useInviteToken.js`